### PR TITLE
Implementing %exit in the test infrasstructure

### PR DESCRIPTION
**Pull request by davidcok**
Adds %exit to the test infrastructure.
A lit command like // RUN: %exit n %dafny ...
where n is a number runs the command after n and tests that its exit code is `n`.

This makes obsolete the need for /countVerificationErrors:0 and dafny_0
and improves on the ! operator to allow tests for specific error codes

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Test/errortests/InvalidArgument1A.dfy [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// RUN: ! %baredafny verify %args --function-syntax:2 null.dfy > "%t"
```
Comment (MikaelMayer) [inline Test/errortests/InvalidArgument1A.dfy:1]: Can't you use %exit here as well?

**Review by MikaelMayer — COMMENTED** [review]
This is great !

Could you also make /countVerificationError parameter explicitly deprecated in the documentation? Also, did you check that there is no more %dafny_0 in the tests?

File: Test/errortests/InvalidArgument1A.dfy (hunk 1 above)
Comment (davidcok) [reply Test/errortests/InvalidArgument1A.dfy:1]: Yes - once this PR is approved, a sequel will roll it out to all the tests.

**Comment by davidcok** [comment]
So far this is just the implementation of %exit -- once approved I will roll it out to all the tests and get rid of /countVerificationErrors:0

**Review by MikaelMayer — APPROVED** [review]
I just fixed a typo in the title for the auto-merge commit. Thanks a lot for making the tests better !

**Pull request merged**